src/vdom.ffi.mjs

```diff
@@ -489,8 +431,15 @@ function getKeyedChildren(el) {
 
   if (el) {
     for (const child of el.children) {
-      const key = child.key || child?.getAttribute("data-lustre-key");
-      if (key) keyedChildren.set(key, child);
+      if (child.elements) {
+        // This is a fragment, the keyed children will be the set of flattened fragment elements
+        iterateElement(child, (fragmentElement) => {
+          if (fragmentElement.key) keyedChildren.set(fragmentElement.key, fragmentElement)
+        });
+      } else {
```

Comment:
This should be simplified as well, fragments' keys arent used to check anything at the moment. I think a fragment should be used to skip the entire sub-tree, but this impl doesn't do that, and I'd want to check behavior of other popular options first

Reply:
pushed a commit which combines logic similarly to other `iterateElement` usage